File: Laba1/Laba1/grille.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Laba1
{
    class grille
    {
        private pig meat;

        private coal coal;

        public pig Meat { set { meat = value; } get { return meat; } }

        public coal Coal { set { coal = value; } get { return coal; } }

        public bool isReady { get { return meat.Has_ready; } }

        public void CookCoal()
        {
            if (coal == null) { return; }
            if (!coal.Has_ready) { coal.Burning(); }
        }

        public void Cook()
        {
            if (!Cheak()) { return; }
            if (!meat.Has_ready) { meat.GetHeat(); }
        }
        private bool Cheak()
        {
            if (coal == null) { return false; }
            if (!coal.Has_ready) { return false; }
            if (meat == null) { return false; }
            return true;
        }
        public bool CoalReady()
        {
            if (coal == null) { return false; }

            if (!coal.Has_ready) { return false; }

            return true;
        }
    }
}

File: Laba1/src/main/Grille.java
package main;

public class Grille {

    private Pig meat;

    private Coal coal;

    public Pig getMeat() {
        return meat;
    }

    public void setMeat(Pig meat) {
        this.meat = meat;
    }

    public Coal getCoal() {
        return coal;
    }

    public void setCoal(Coal coal) {
        this.coal = coal;
    }

    public boolean isReady(){
        return meat.isReady();
    }

    public void cookCoal(){
        if(coal==null){
            return;
        }
        if(!coal.isReady()){
            coal.burning();
        }
    }
    public void cook(){
        if(!cheak()){
            return;
        }
        if(!meat.isReady()){
            meat.GetHeat();
        }
    }
    private boolean cheak(){
        if(coal==null){
            return false;
        }
        if(!coal.isReady()){
            return false;
        }
        if(meat==null){
            return false;
        }
        return true;
    }
    public boolean coalReady(){
        if(coal==null){
            return false;
        }
        if(!coal.isReady()){
            return false;
        }
        return true;
    }
}
